src/canvas/PatternPaths/PatternPath.tsx

```diff
@@ -135,6 +162,23 @@ export abstract class PatternPath implements IPatternPath {
         return this._fittedSegment;
     };
 
-    abstract setFittedSegment(): void;
+    setFittedSegment = (): void => {
+        this._setFittedSegment();
+        this._points = this._fittedSegment ? this._fittedSegment.computePoints() : new Array<Point>();
+    };
+
+    splitAtPoint = (interesection: Point): PatternPath[] => {
+        const originalSegment = this.getSegment();
+        const splitSegments = originalSegment.split(interesection);
+        if (splitSegments.length !== 2) {
+            throw new Error("Split did not return correct number of segments");
+        }
+
+        return this._createPathFromSegment(splitSegments);
+    };
+
+    protected abstract _createPathFromSegment(segments: Segment[]): PatternPath[];
```

Comment:
maybe rename this to _createPathsFromSegments?

Reply:
Sure, I changed it.